src/main/java/uk/gov/hmcts/reform/civil/service/robotics/mapper/EventHistoryMapper.java

```diff
@@ -495,6 +500,33 @@ private void buildBreathingSpaceEvent(EventHistory.EventHistoryBuilder builder,
         }
     }
 
+    private void buildCoscEvent(EventHistory.EventHistoryBuilder builder, CaseData caseData) {
+        // CoSCApplicationStatus can be null if the claimant has mark the case as paid in full before defendant applies for a cosc
+        if (featureToggleService.isJOLiveFeedActive()
+            && featureToggleService.isCoSCEnabled()
+            && caseData.hasCosc()) {
```

Comment:
Does this mean that certificate is already generated? Image scenario that certificate is requested and add case note event happen-> This will add event but certificate was not still generated